src/hooks/useAuthorityLinking/useAuthorityLinking.js

```diff
@@ -242,6 +251,53 @@ const useAuthorityLinking = () => {
     };
   }, []);
 
+  const actualizeNewLinkedFields = useCallback(async (formValues) => {
+    const isNewLinkedField = (field) => field.linkDetails?.status === AUTOLINKING_STATUSES.NEW;
+    const hasNewLinkedField = formValues.fields.some(isNewLinkedField);
+
+    if (!hasNewLinkedField) {
+      return formValues;
+    }
+
+    const newLinkedFields = formValues.fields.filter(isNewLinkedField);
+    const payload = hydrateForLinkSuggestions(formValues, newLinkedFields);
+    const { fields: suggestedFields } = await fetchLinkSuggestions({
+      body: payload,
+      isSearchByAuthorityId: true,
+    });
+
+    let suggestedFieldIndex = 0;
+
+    const actualizedFields = formValues.fields.map(field => {
+      if (isNewLinkedField(field)) {

```

Comment:
Let's use fail-fast approach: check for negative cases that don't need special treatment first:
```js
if (!someCondition) {
  return field;
}

// do some more actions here
```

It's going to reduce code complexity by making it less indented